examples/sysaudio/main.zig

```diff
@@ -72,9 +79,20 @@ pub const ToneEngine = struct {
     pub fn render(engine: *ToneEngine, descriptor: sysaudio.DeviceDescriptor, buffer: []u8) void {
         // TODO(sysaudio): demonstrate how to properly handle format of the buffer here.
         // Right now we blindly assume f32 format, which is wrong (but always right in WASM.)
-        const sample_rate = @intToFloat(f32, descriptor.sample_rate.?);
-        const buf = @ptrCast([*]f32, @alignCast(@alignOf(f32), buffer.ptr))[0 .. buffer.len / @sizeOf(f32)];
-        const frames = buf.len / descriptor.channels.?;
+        engine.sample_rate = @intToFloat(f32, descriptor.sample_rate);
+        switch (descriptor.format) {
+            .U8 => renderWithType(u8, engine, descriptor, buffer),
+            .S16 => renderWithType(i16, engine, descriptor, buffer),
+            .S24 => renderWithType(i24, engine, descriptor, buffer),
+            .S32 => renderWithType(i32, engine, descriptor, buffer),
+            .F32 => renderWithType(f32, engine, descriptor, buffer),
+        }
+    }
+
+    pub fn renderWithType(comptime T: type, engine: *ToneEngine, descriptor: sysaudio.DeviceDescriptor, buffer: []u8) void {
```

Comment:
I wonder why this function should this not be receiving a typed buffer? Seems odd to handle that inside the function rather than at the call site

Reply:
I just hadn't thought of it! I've implemented your suggestion